File: programmers/기초/특별한 이차원 배열 1.js
//특별한 이차원 배열 1
// arr[i][j] (0 ≤ i, j < n)의 값은 i = j라면 1, 아니라면 0입니다.
function solution(n) {
  let array = Array(n).fill(Array(n).fill(0));

  return array.map((arr, index) => {
    return arr.map((a, idx) => (idx === index ? 1 : a));
  });
}

File: programmers/python/특별한 이차원 배열 1.py
# 특별한 이차원 배열 1
def solution(n):
    result = []
    for i in range(n):
        arr = [0] * n
        arr[i] = 1
        result.append(arr)
    return result



def solution(n):
    answer=[[0]*n for i in range(n)]
    for i in range(n): answer[i][i]=1
    return answer